dll/cpl/timedate/internettime.c

```diff
@@ -319,6 +351,41 @@ OnInitDialog(HWND hwnd)
     CreateNTPServerList(hwnd);
 }
 
+static VOID
+OnAutoSync(BOOL Sync)
+{
+    HKEY hKey;
+    LONG lRet;
+    WCHAR szAuto[7];
+
+    if (Sync)
+        StringCbCopy(szAuto, wcslen(L"NTP\0"), L"NTP\0");
```

Comment:
This type of construct doesn't work (aside from the comment above about the buffer size parameter). String copy functions will stop at a null char, so the extra \0 has no effect. If you need a double-null (e.g. for REG_MULTI_SZ), you need to write it manually -- however it doesn't seem like you do. Regular strings are already guaranteed to end with a null.

Reply:
I will take out the "\0" entries. My habit is to be extra sure and safe, so this is represented here. Thanks.